building/glass/upload.py

```diff
@@ -122,20 +121,27 @@ def gs_rsync(local_path: str, remote_path: str, boto_path: str):
 """.lstrip()
 
 
-def perform_uploads(uploads: dict, branch: str) -> None:
-    if not os.path.exists(BOTO_PATH):
-        raise Exception("you need to put the GCP service account private key file into /homeworld/boto-key")
+def perform_uploads(uploads: dict, branch_config: aptbranch.AptBranchConfig) -> None:
     with tempfile.TemporaryDirectory() as staging:
         root = os.path.join(staging, "root")
         for remote_path, local_path in uploads.items():
             target = os.path.join(root, remote_path.lstrip("/"))
             if not os.path.isdir(os.path.dirname(target)):
                 os.makedirs(os.path.dirname(target))
             shutil.copy2(local_path, target)
-        botoconfig_name = os.path.join(staging, "boto.config")
-        with open(botoconfig_name, "w") as bout:
-            with open(BOTO_PATH, "r") as f:
-                project_id = json.load(f)["project_id"]
-            bout.write(BOTO_TEMPLATE % (BOTO_PATH, project_id))
-            bout.flush()
-        gs_rsync(root, branch, botoconfig_name)
+
+        upload_method = branch_config.upload_config["method"]
+        if upload_method == "google-cloud-storage":
+            if not os.path.exists(BOTO_PATH):
+                raise Exception("you need to put the GCP service account private key file into /homeworld/boto-key")
+            botoconfig_name = os.path.join(staging, "boto.config")
+            with open(botoconfig_name, "w") as bout:
+                with open(BOTO_PATH, "r") as f:
+                    project_id = json.load(f)["project_id"]
+                bout.write(BOTO_TEMPLATE % (BOTO_PATH, project_id))
+                bout.flush()
+            gs_rsync(root, branch_config.name, botoconfig_name)
+        elif upload_method == "rsync":
```

Comment:
This should dispatch to different methods, rather than using an if/else sequence.